### PR TITLE
Redeploy deployed version

### PR DESCRIPTION
**Pull request by gregbiv**
by providing ~-e deployment_force=true~ `-e deployment_overwrite=true` it will overwrite the already deployed version
 
- [x] fix issue #43 
- [ ] add tests

### CODE REVIEW

**Comment by kieranajp** [comment]
I would suggest we rename the flag as everyone just copy pastes this one. Perhaps deployment_overwrite.

**Comment by kieranajp** [comment]
Lgtm apart from tests, second opinion anyone?

**Comment by gregbiv** [comment]
test are failed because of funny reason `self.repository.release_from_tag(self.version)\r\nAttributeError: 'Repository' object has no attribute` :notme:

**Comment by olvlvl** [comment]
Please update the documentation. We must make this very clear.

**Comment by boekkooi-fresh** [comment]
Should we even have a difference between `deployment_force` and `deployment_override`? It seems to me that `deployment_force` should always override the deployment.

**Comment by kieranajp** [comment]
@boekkooi-fresh Normally I'd agree, but in practice everyone uses deployment_force all the time, which effectively makes overwriting the deployment the default.

I'd consider deployment_force deprecated at this point in favour of deployment_override.

**Comment by gregbiv** [comment]
but it will affect only people who will try to deploy already deployed version

**Comment by olvlvl** [comment]
@kieranajp We should educate people instead of coming with work-around for their bad behavior.

**Comment by kieranajp** [comment]
It's not about bad behaviour at this point, it's about it being coded into [config](https://github.com/hellofresh/intfood/blob/b4194de3140fb48b9dc31ac3b240ddb7705bdbef/config/build/vars/deployment.yml#L5) everywhere.

If we set `deployment_force` to actually overwrite the deployment, then even if people don't use `deployment_force` from the CLI, often fpm, consumers, etc will be restarted anyway. Do we want to restart everything every 4 hours when policy enforcement triggers? I doubt that.

As for what `deployment_force` _does_: it only force recreates config files; it doesn't redeploy the artefact. This is not really obvious behaviour at all. My suggestion of using `deployment_override` isn't meant to be _in addition_ to `deployment_force` - I mean to _remove_ deployment_force altogether (making it a noop). This is because it's actually pretty useless, and poorly named. I want to be more explicit in the naming.

**Comment by boekkooi-fresh** [comment]
I'm also not sure `override` is the correct phrase maybe `deployment_redeploy` is better?

**Comment by gregbiv** [comment]
@boekkooi-fresh it's `overwrite` :trollface:

File: README.md [hunk 1]
```diff
@@ -147,8 +147,11 @@ deployment_post                     : False
 deployment_cron_jobs                : "none"
 
 ## By default you deploy once and to override you must pass true to force deployment
-deployment_force                    :  false
-deployment_guard_file               :  "/var/local/deployment_first_boot_file"
+deployment_force                    : False
+deployment_guard_file               : "/var/local/deployment_first_boot_file"
+
+## By default you deploy once but if you want to overwrite already deployed draft you need to force deployment
+deployment_overwrite                : False
```
Comment (wellfreire) [inline README.md]: I'd just improve the comments for both `deployment_force` and `deployment_overwrite`, they seem a bit confusing 🤔 .

If I correctly understood, we realised now that what `deployment_force` actually does is simply forcing config to be re-written. The new flag now will do the real job of forcing/overwriting the deployment.

Let's just make sure that's clear in those comments.

**Comment by kieranajp** [comment]
Additionally, upon further thought, changing `deployment_force` to actually force a redeploy will break all the idempotency tests 😅  so having a flag with a different name is the only way

**Comment by rnurgaliyev** [comment]
Any update here?

**Comment by kieranajp** [comment]
:shipit:

**Pull request merged**